Dockerfile-api

```diff
@@ -1,14 +1,19 @@
+# Build container
 FROM golang:1.17 AS buildContainer
 WORKDIR /go/src/app
-
 COPY . .
-
 #flags: -s -w to remove symbol table and debug info
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -mod mod -ldflags "-s -w" -o ./templatesapi ./cmd/templates
 
+# SSL certificates container
+FROM alpine as sslCerts
+RUN apk add -U --no-cache ca-certificates
+
+# Production container
 FROM scratch
-WORKDIR /app
+COPY --from=sslCerts /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```

Comment:
Nevedel som sa pripojit na Storage API, pretoze v image neboli ziadn root certs.
SSL je build-in v binarke, ale certifikaty je potrebne dodat.

Alpine ich ma v jednom subore, tak som to vyuzil.

Reply:
btw. zrovna ten scratch image měním na něco jinýho protože potřebujeme ten `git` command 🙂

Reply:
jj, ta alpine by nebola zla, maly image ...

Reply:
@JakubMatejka inak co tak staticky skompilovany `git` :trollface:

Reply:
🤔 mno, tady, zdá se, dělali něco podobnýho 😃: https://github.com/rancher/docker-from-scratch/blob/master/Dockerfile (ale je to archivovaný).

Zkusil jsem https://github.com/bitnami/minideb na kterým jsem debugoval tu první verzi a ten má navíc 50 MB plus 77 MB git. 
![CleanShot 2022-03-15 at 13 56 40](https://user-images.githubusercontent.com/215660/158382781-66aaa8fc-52f9-4f51-abdc-30ab577d4dba.jpg)

Reply:
vo vysledku je to jedno, ... `alpine` pri pride ako standard pre tieto optimalizovane images, ale aj `minideb` moze byt OK.
Hlavny rozdiel bol obycajne v tom, ze `alpine` balicky nemaju dokumentaciu a iny balast, a debian maju.
S `minideb` som sa este nestretol, tak neviem, nema nejak extra vela :star: `1.6k` a neprechadzaju im aktualne CI builds, hlasoval by som skor za `alpine`, staci pouzit `RUN apk add -U --no-cache ca-certificates git` a mas to (a zmaza `FROM scratch`, necham to na teba, co ti pride lepsie.

Reply:
Alpine `git` ma `12Mb`: https://pkgs.alpinelinux.org/package/edge/main/x86/git

Reply:
jo, ale problem nebol :) to bolo este pred kompilaciou `Go` ... a suviselo s tou `not found` chybou, ... tj. `apline` nepouziva `libc` ale `musl`, ina implementacia `c runtime` .... ale nasa apka je uplne staticka, tam uz problem nie je, ... a `git` balicek v ich repe je kompilovany s `musl`, takze ziaden problem.

Reply:
A jo, ty už jsi psal. Nepřišly mi notifikace. 🙂 Ok, zkusil bych ten Alpine. 👍